javascript/javascript1/week2/homework-js1-w2.js

```diff
@@ -0,0 +1,168 @@
+//*****************************************/
+const currentDate = new Date();
+const todayIndex = currentDate.getDay();
+const weekdays = [
+  "Sunday",
+  "Monday",
+  "Tuesday",
+  "Wednesday",
+  "Thursday",
+  "Friday",
+  "Saturday",
+];
+
+function getEventWeekday(daysFromToday) {
+  const eventDay = weekdays[(todayIndex + daysFromToday) % 7];
+  return eventDay;
+}
+
+console.log(getEventWeekday(2));
+
+//*****************************************/
+// Weather wear
+//*****************************************/
+function getClothes(temp) {
+  if (temp <= -20) {
+    return "It's too cold, please do not go out.";
+  } else if (temp <= -5) {
+    return "Down jacket, warm scarf, and gloves";
+  } else if (temp <= 10) {
+    return "Heavier coat, sweater or hoodie, and jeans";
+  } else if (temp <= 20) {
+    return "Light jacket, light jeans";
+  } else if (temp <= 40) {
+    return "Skirt, shorts and a t-shirt";
+  } else if (40 < temp) {
+    return "It's too hot, please do not go out.";
+  }
+}
+
+const clothesToWear = getClothes(18);
+console.log(clothesToWear);
+
+//*****************************************/
+// Student manager
+//*****************************************/
+const class07Students = [];
+function addStudentToClass(studentName) {
+  // You write code here
+  const lowercaseClass07Students = class07Students.map((name) =>
+    name.toLowerCase()
+  );
+  if (typeof studentName !== "string" || studentName === "") {
+    console.log("Please provide a valid name.");
+  } else if (lowercaseClass07Students.includes(studentName.toLowerCase())) {
+    console.log(`Student ${studentName} is already in the class`);
+  } else if (getNumberOfStudents() <= 5) {
+    class07Students.push(studentName);
+    console.log(`${studentName} has been successfully added.`);
+  } else if (studentName === "Queen") {
+    class07Students.push(studentName);
+    console.log(`${studentName} has been successfully added.`);
+  } else {
+    console.log("Cannot add more students to class 07");
+  }
+}
+
+function getNumberOfStudents() {
+  // You write code here
+  return class07Students.length;
+}
+
+addStudentToClass("Benjamin");
+addStudentToClass("Alice");
+addStudentToClass("Bob");
+addStudentToClass("Julia");
+addStudentToClass("Jacob");
+addStudentToClass("Amy");
+addStudentToClass("John"); // Add more students than there is space for.
+addStudentToClass(""); // Try to add an empty string.
+addStudentToClass("Benjamin"); // Add a student that is already in the class.
+addStudentToClass("Queen"); // Add the Queen to a full class.
+console.log(getNumberOfStudents());
+
+//*****************************************/
+// Candy helper optional
+//*****************************************/
+const candyPrice = {
+  "sweet": 0.5,
+  "chocolate": 0.7,
+  "toffee": 1.1,
+  "chewing-gum": 0.03,
+};
+const boughtCandyPrices = [];
+
+function addCandy(candyType, weight) {
+  const calculateCandyPrice = candyPrice[candyType.toLowerCase()] * weight;
```

Comment:
Does it still work if you remove the `toLowerCase()`?

Reply:
Sure it works, but I prefer to keep it. Since we do not need to consider uppercase or lowercase when entering input. Maybe too much for this task, but just for practice 😊.